packages/contract-wrappers/src/contract_wrappers/coordinator_wrapper.ts

```diff
@@ -0,0 +1,672 @@
+    ): Promise<string> {
+        if (orderTransactionOpts.shouldValidate) {
+            await this._contractInstance.executeTransaction.callAsync(
+                transaction,
+                txOrigin,
+                transactionSignature,
+                approvalExpirationTimeSeconds,
+                approvalSignatures,
+                {
+                    from: txOrigin,
+                    gas: orderTransactionOpts.gasLimit,
+                    gasPrice: orderTransactionOpts.gasPrice,
+                    nonce: orderTransactionOpts.nonce,
+                },
+            );
+        }
+        const txHash = await this._contractInstance.executeTransaction.sendTransactionAsync(
+            transaction,
+            txOrigin,
+            transactionSignature,
+            approvalExpirationTimeSeconds,
+            approvalSignatures,
+            {
+                from: txOrigin,
+                gas: orderTransactionOpts.gasLimit,
+                gasPrice: orderTransactionOpts.gasPrice,
+                nonce: orderTransactionOpts.nonce,
+            },
+        );
+        return txHash;
+    }
+
+    private async _toSignedZeroExTransactionAsync(
+        data: string,
+        senderAddress: string,
+    ): Promise<SignedZeroExTransaction> {
+        await assert.isSenderAddressAsync('senderAddress', senderAddress, this._web3Wrapper);
+        const normalizedSenderAddress = senderAddress.toLowerCase();
+
+        const transaction: ZeroExTransaction = {
+            salt: generatePseudoRandomSalt(),
+            signerAddress: normalizedSenderAddress,
+            data,
+            verifyingContractAddress: this.exchangeAddress,
+        };
+        const transactionHash = transactionHashUtils.getTransactionHashHex(transaction);
+        const signature = await signatureUtils.ecSignHashAsync(
+            this._web3Wrapper.getProvider(),
+            transactionHash,
+            transaction.signerAddress,
+        );
+        return {
+            ...transaction,
+            signature,
+        };
+    }
+
+    private async _requestServerResponseAsync(
+        transaction: SignedZeroExTransaction,
+        senderAddress: string,
+        feeRecipientAddress: string,
+    ): Promise<CoordinatorServerApprovalResponse | CoordinatorServerCancellationResponse> {
+        assert.doesConformToSchema('transaction', transaction, schemas.zeroExTransactionSchema);
+        assert.isETHAddressHex('feeRecipientAddress', feeRecipientAddress);
+
+        const coordinatorOperatorEndpoint = await this._registryInstance.getCoordinatorEndpoint.callAsync(
+            feeRecipientAddress,
+        );
+        if (coordinatorOperatorEndpoint === undefined) {
+            throw new Error(
+                `Could not find endpoint for coordinator operator with { coordinatorAddress: ${
+                    this.address
+                }, registryAddress: ${this.registryAddress}, operatorAddress: ${feeRecipientAddress}}`,
```

Comment:
This error message feels a bit confusing to me. How about:

```
throw new Error(`No Coordinator server endpoint found in Coordinator registry for feeRecipientAddress: ${feeRecipientAddress}. Registry contract address: ${this.registryAddress}`);
```

Reply:
this sounds way better, thanks!